RecoEgamma/EgammaTools/src/ShowerDepth.cc

```diff
@@ -0,0 +1,30 @@
+#include "RecoEgamma/EgammaTools/interface/ShowerDepth.h"
+#include <cmath>
+
+using namespace hgcal;
+
+float ShowerDepth::getClusterDepthCompatibility(float measuredDepth, float emEnergy, float& expectedDepth, float& expectedSigma) const {
+    double lny = (emEnergy > criticalEnergy_) ? std::log(emEnergy/criticalEnergy_) : 0.;
```

Comment:
it seems like all of this function should be float.Or is there a concern about loss of precision someplace?